src/store/src/Document/Loader.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\AI\Store\Document;
+
+/**
+ * @author Christopher Hertel <mail@christopher-hertel.de>
+ */
+final class Loader
+{
+    /**
+     * @var SourceTypeLoaderInterface[]
+     */
+    private readonly array $sourceTypeLoaders;
+
+    /**
+     * @param iterable<SourceTypeLoaderInterface> $sourceTypeLoaders
+     */
+    public function __construct(
+        iterable $sourceTypeLoaders,
+    ) {
+        $this->sourceTypeLoaders = $sourceTypeLoaders instanceof \Traversable ? iterator_to_array($sourceTypeLoaders) : $sourceTypeLoaders;
```

Comment:
Why do you break lazyness ? I saw theses bits in other place, and I don't get it
It's okay to store iterable in a class: https://3v4l.org/d6h65